### DISCUSSION TITLE
How to target SDK 34 ?

### DISCUSSION THREAD

**Discussion by stephanberger4** [airsdk/Adobe-Runtime-Support#3337] [Q&A] [answered]
Hi, 

Google play is now requiring to target SDK 34 (Android 14). I've download the last AIR SDK and compile my app but it still targeting SDK 33. 
How to target SDK 34 ? 

Thank you

**Comment by marchbold** [comment]
Add the following to your manifest additions (adjust minimum version as needed):

```xml 
<uses-sdk android:minSdkVersion="19" android:targetSdkVersion="34" />
```

**Reply by ajwfrost** [reply, marked as answer]
There's a handy guide that someone wrote :-)
https://medium.com/airnativeextensions/adobe-air-and-the-android-target-sdk-59f5b60bb9bd

FYI we will be setting targetSdkVersion of 34 to be the new default, in the 51.1 release...

**Reply by sjabberwocky** [reply]
> FYI we will be setting targetSdkVersion of 34 to be the new default, in the 51.1 release...
@ajwfrost Do you have a release window for this update? When do you expect to release it?

**Reply by ajwfrost** [reply]
For 51.1: I am hoping by the end of this month. There are a couple more items going in plus we need to do all the template updating, but hopefully we can finish all that next week...

**Reply by NoodleKidz** [reply]
Thanks, waiting for this update, you guys are awesome.

Edit: The AIR update is out, it does target SDK 34, now my app's update is being reviewed by Google Play Console.

Edit2: The app update is approved by Google.

**Reply by BigPixelTechnologies** [reply]
When I tried publishing my app with that SDK 51.1, I get the following warning
![image](https://github.com/user-attachments/assets/32ad226a-c76f-4667-ac01-ba35b9093156)
Although I have set the target sdk to 34 <uses-sdk android:minSdkVersion="19" android:targetSdkVersion="34"/> and
I have installed the sdk 34 in android studio as well.
**Moreover I see that the app is lagging when using the latest sdk 51.1**, which does not happen if I switch back to old sdk v50.2

**Reply by GeneralVimes** [reply]
Yes, there is nothing to worry, however, you can have multiple build-tools, but it's better to make sure your newest build-tool is 34.0.0

For this open Android Studio - SDK manager - SDK Tools - and select a checkbox next to 34.0.0

**Comment by stephanberger4** [comment]
Than you very much for your answer !

Cheers !
________________________________
De : Michael ***@***.***>
Envoyé : dimanche 7 juillet 2024 16:22
À : airsdk/Adobe-Runtime-Support ***@***.***>
Cc : stephanberger4 ***@***.***>; Author ***@***.***>
Objet : Re: [airsdk/Adobe-Runtime-Support] How to target SDK 34 ? (Discussion #3337)


Add the following to your manifest additions (adjust minimum version as needed):

<uses-sdk android:minSdkVersion="19" android:targetSdkVersion="34" />

**Comment by bontegames** [comment]
All my games got a policy violation in Google Play as well, stating I need to target API level 34 before August 31.
I'm a bit confused by your answer. Does it mean I can already target 34 with current AIR, or do I wait for the 51.1 release?
If I have to wait for 51.1, when can it be expected? Thanks!

**Reply by marchbold** [reply]
You can already target 34, just make the change mentioned in your manifest

**Reply by FliplineStudios** [reply]
@bontegames I got a bunch of those warnings too.  The messages Google sends are worded poorly, but for what it's worth it looks like as long as you're currently targeting API level 33 for existing apps they'll still be compliant, but you'll need to target 34 for any app updates to them after August 30th (and will need to at least target 34 before August 2025 to stay discoverable in the store).

https://support.google.com/googleplay/android-developer/answer/11926878

**Comment by josquinet** [comment]
I would like to ask a question, I use animate and air sdk harmann...when creating my aab file in android studio/android sdk, I have API 33 installed. If now I change in my manifest to point to API 34, when creating my updates I have to do it by installing API 34 in android studio to create the aab or it is not necessary and I can continue creating the aab with API 33? thank you very much.

**Reply by ajwfrost** [reply]
Hi

I just opened an Android Studio project and changed the `targetSdkVersion` value to 36, and it built okay. I think it's the `compileSdkVersion` and `buildToolsVersion` that specify versions of the actual SDKs/build tools that need to be present on the computer. And for AIR builds, those are picked up automatically.

But it seems like it's fine to compile with an old version of the Android SDK and tools, and yet target a newer API level.

thanks

**Comment by HanSpyMB** [comment]
Good morning,

Could someone explain to me step by step how to force the sdk to target API 34.
I'm exporting a bundle from animate and I'm a little lost.
All my exports target version 33 and I don't understand how to change it despite reading the link given above. I'll give you a screenshot of my android studio.
I have the latest version of AIR SDK 51.0.1.4
<img width="971" alt="Capture d’écran 2024-07-29 à 16 21 37" src="https://github.com/user-attachments/assets/5bbe35f9-d0ac-4016-bafa-0caa901caade">

**Reply by marchbold** [reply]
Did you update the target sdk version in your manifest additions?

**Reply by ajwfrost** [reply]
Hopefully step-by-step this would be:
1) Close Animate
2) Open the file ProjectName-app.xml that should be next to the FLA file, in a text editor
3) Find your `manifestAdditions` section. If you don't see this, within an `<android>` block, then you can add it per the below.
4) Add the target SDK request into this, in a `uses-sdk` tag, with the `targetSdkVersion` set to 34. See below example.
5) Save the file and then re-open Animate.

```
    <android>
        <manifestAdditions><![CDATA[
            <manifest>
                <uses-sdk android:targetSdkVersion="34"/>
            </manifest>
        ]]></manifestAdditions>
    </android>
````

**Reply by HanSpyMB** [reply]
@ajwfrost Thanks
I just tried without success.
It regenerates the xml and deletes the tag

**Reply by HanSpyMB** [reply]
I have found the solution : 

Select the checkbox "Manually manage permissions and manifest..." and click OK, then you will see a .xml file appear right beside your .fla file, open it with any text editor and copy-paste manifest permissions you have into that file, you will see android and manifestAdditions tags, just replace them with one you have, save and recompile your fla, done!

**Reply by josquinet** [reply]
to complement the screenshot I have seen of the android sdk, , In the release of the harmann version you use the android sdk target is 33. I use that one to create the aab file. What I do (and it works for me) is to open show package details, install all the packages from sdk 33 and then create the aab. It doesn't matter that I have set sdk 33 as target in your app manifest file.

**Reply by josquinet** [reply]
It doesn't matter that I have set sdk 34* as target in your app manifest file.

**Comment by GeneralVimes** [comment]
Google said to me that "A recent policy violation in your app has been addressed".
This is what I did.

In Android Studio in SDK manager I selected in SDK Platforms: Android SDK Platform 34, Sources for Android 34, Android SDK Platform 34-ext8 and Android SDK Platform 34-ext12. Also I unselected the checkboxes which I had near SDK34 there. Also I selected 34.0.0 in SDK Tools. Then the needed Android SDKs were downloaded and installed.
I updated AIR SDK to 51.0.1.4
I installed Java 17.0.2 from OpenJDK and then I set a path to it in adt.cfg file, like: JAVA_HOME=C:\Program Files\Java\jdk-17.0.
This is what was needed to do with the tools. Then with the project I just had to update the line in application.xml, like:
<uses-sdk android:minSdkVersion="16" android:targetSdkVersion="34"/>

I also updated all the Distriqt's ANEs to get the In-App version of 6.2.1

I compiled .aab and released it on Google Play and something like one day after that I saw a message in the developer console: "A recent policy violation in your app has been addressed".
